backend/src/main/java/com/bakdata/conquery/apiv1/MetaDataPatch.java

```diff
@@ -54,14 +55,15 @@
 			.accept(this);
 	}
 	
-	protected <T extends MetaDataPatch, ID extends IId<?>, INST extends Taggable & Shareable & Labelable & Identifiable<? extends ID>> Consumer<T> buildChain(Consumer<T> patchConsumerChain,MetaStorage storage, User user, INST instance, PermissionCreator<ID> permissionCreator){
-		if(getTags() != null && user.isPermitted(permissionCreator.apply(Ability.TAG.asSet(), instance.getId()))) {
+	protected <T extends MetaDataPatch, ID extends IId<?>, INST extends Taggable & Shareable & Labelable & Identifiable<? extends ID>& Owned> Consumer<T> buildChain(Consumer<T> patchConsumerChain,MetaStorage storage, User user, INST instance, PermissionCreator<ID> permissionCreator){
+		boolean owned = instance.getOwner().equals(user.getId());
+		if(getTags() != null && (owned || user.isPermitted(permissionCreator.apply(Ability.TAG.asSet(), instance.getId())))) {

```

Comment:
wäre es nicht sinnvoll diesen check nach User#isPermitted zu schieben?

Reply:
`boolean owned = instance.getOwner().equals(user.getId());`

Reply:
`isPermitted` ist eine Funktion von Shiro. Die gibt die Permissions einfach in den Security.Manager weiter

Reply:
Und das zu überschreiben im User?

Reply:
Aber das wäre eine andere Funktion, das wäre nicht mehr 'isPermitted(Permission/Collection<Permission>/...)' sondern 'isOwn(Owned), oder?

Reply:
ah, sorry ich hab mir das interface anders vorgestellt. du hast recht das geht maximal nur so.